src/libraries/System.Runtime.InteropServices.JavaScript/gen/JSImportGenerator/JSImportStubContext.cs

```diff
@@ -63,62 +34,22 @@ static bool IsSkipLocalsInitAttribute(AttributeData a)
-                // Remove current members, attributes, and base list so we don't double declare them.
-                typeDecl = typeDecl.WithMembers(List<MemberDeclarationSyntax>())
-                                   .WithAttributeLists(List<AttributeListSyntax>())
-                                   .WithBaseList(null);
-
-                containingTypes.Add(typeDecl);
-
-                stubTypeFullName = currType.Name + (string.IsNullOrEmpty(stubTypeFullName) ? "" : ".") + stubTypeFullName;
-                currType = currType.ContainingType;
-            }
-            stubTypeFullName = stubTypeNamespace == null ? stubTypeFullName : (stubTypeNamespace + "." + stubTypeFullName);
-
-            (ImmutableArray<TypePositionInfo> typeInfos, IMarshallingGeneratorFactory generatorFactory) = GenerateTypeInformation(method, diagnostics, env);
-
-            ImmutableArray<AttributeListSyntax>.Builder additionalAttrs = ImmutableArray.CreateBuilder<AttributeListSyntax>();
+            ImmutableArray<IUseSiteAttributeParser> useSiteAttributeParsers = ImmutableArray.Create<IUseSiteAttributeParser>(new JSMarshalAsAttributeParser(env.Compilation));
+            var jsMarshallingAttributeParser = new MarshallingInfoParser(
+                diagnostics,
+                new MethodSignatureElementInfoProvider(env.Compilation, diagnostics, method, useSiteAttributeParsers),
+                useSiteAttributeParsers,
+                ImmutableArray.Create<IMarshallingInfoAttributeParser>(new JSMarshalAsAttributeParser(env.Compilation)),
+                ImmutableArray.Create<ITypeBasedMarshallingInfoProvider>(new FallbackJSMarshallingInfoProvider()));
+            SignatureContext sigContext = SignatureContext.Create(method, jsMarshallingAttributeParser, env, typeof(JSImportGenerator).Assembly);
 
-            if (env.TargetFramework != TargetFramework.Unknown)
-            {
-                // Define additional attributes for the stub definition.
-                additionalAttrs.Add(
-                    AttributeList(
-                        SingletonSeparatedList(
-                            Attribute(ParseName(TypeNames.System_CodeDom_Compiler_GeneratedCodeAttribute),
-                                AttributeArgumentList(
-                                    SeparatedList(
-                                        new[]
-                                        {
-                                            AttributeArgument(LiteralExpression(SyntaxKind.StringLiteralExpression, Literal(GeneratorName))),
-                                            AttributeArgument(LiteralExpression(SyntaxKind.StringLiteralExpression, Literal(GeneratorVersion)))
-                                        }))))));
-            }
+            string stubTypeFullName = method.ContainingType.ToDisplayString(new SymbolDisplayFormat(typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces));
```

Comment:
`NameAndContainingTypesAndNamespaces` is better than what we have in the `GetFullyQualifiedMethodName` and `s_typeNameFormat`, please unify it.

Reply:
If we use `NameAndContainingTypesAndNamespaces` in `GetFullyQualifiedMethodName`, how do we determine which "." separate class names and which namespace names?

Context: mono special naming convention requires to separate class names with `/` and namespaces with '.'.

Reply:
I think to meet the Mono naming requirements with `NameAndContainingTypesAndNamespaces` would require us to go back to using `ToDisplayParts()`. I'll leave that for now since we just moved away from that.